operator/v1/types_dns.go

```diff
@@ -86,19 +86,40 @@ type Server struct {
 	ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
 }
 
+// ForwardingPolicy is the policy to use when forwarding DNS requests.
+// +kubebuilder:validation:Enum=Random;RoundRobin;Sequential
+type ForwardingPolicy string
+
+const (
+	// RandomForwardingPolicy picks a random upstream server to query
+	RandomForwardingPolicy ForwardingPolicy = "Random"
+
+	// RoundRobinForwardingPolicy picks upstream servers in a round-robin order
+	RoundRobinForwardingPolicy ForwardingPolicy = "RoundRobin"
+
+	// SequentialForwardingPolicy tries querying upstream servers in a sequential order
+	SequentialForwardingPolicy ForwardingPolicy = "Sequential"
+)
+
 // ForwardPlugin defines a schema for configuring the CoreDNS forward plugin.
 type ForwardPlugin struct {
 	// upstreams is a list of resolvers to forward name queries for subdomains of Zones.
-	// Upstreams are randomized when more than 1 upstream is specified. Each instance of
-	// CoreDNS performs health checking of Upstreams. When a healthy upstream returns an
-	// error during the exchange, another resolver is tried from Upstreams. Each upstream
-	// is represented by an IP address or IP:port if the upstream listens on a port other
-	// than 53.
+	// Each instance of CoreDNS performs health checking of Upstreams. When a healthy upstream
+	// returns an error during the exchange, another resolver is tried from Upstreams. The
+	// Upstreams are selected in the order specified in Policy. Each upstream is represented
+	// by an IP address or IP:port if the upstream listens on a port other than 53.
 	//
 	// A maximum of 15 upstreams is allowed per ForwardPlugin.
 	//
 	// +kubebuilder:validation:MaxItems=15
 	Upstreams []string `json:"upstreams"`
+
+	// policy is used to determine the order in which upstream servers are selected for querying.
+	// The default value is "Random"
+	//
+	// +optional
+	// +kubebuilder:default:="Random"
```

Comment:
This default surprised me a little bit and led me to do some archaeology (feel free to skip the following excursion if you aren't interested).  

The original enhancement proposal for adding the API to allow specifying custom upstream resolvers actually included an API field to allow specifying the forwarding policy as well as some discussion about what the default policy should be (see https://github.com/openshift/enhancements/pull/53#discussion_r329703046), but somehow that API field didn't make it into the API (https://github.com/openshift/api/pull/441), so the implementation just went with CoreDNS's default of "Random" (https://github.com/openshift/cluster-dns-operator/pull/139).  

We did, however, deliberately change the policy for upstream resolvers from `/etc/resolv.conf` to be "Sequential" for consistency with documented `/etc/resolv.conf` semantics (https://github.com/openshift/cluster-dns-operator/pull/129).  

It is kind of weird that the default policy for upstream resolvers specified in the DNS API object is "Random" while the policy for upstream resolvers specified in `/etc/resolv.conf` is "Sequential".  The policy for `/etc/resolv.conf` definitely should be "Sequential", and it probably isn't worth the risk to change the default for servers specified in the API object, but the discrepancy is odd.